### PR TITLE
amp-form mustache template rendering doc height update

### PR DESCRIPTION
**Pull request by alabiaga**
When the amp-form renders mustache templates, the mutateElement call needs to occur to update the viewer of the doc height changes that might occur as a result of the rendering.

b115564565

### CODE REVIEW

File: extensions/amp-form/0.1/amp-form.js [hunk 1]
```diff
@@ -1052,14 +1052,13 @@ export class AmpForm {
                   /* detail */ null,
                   {bubbles: true});
               container.dispatchEvent(renderedEvent);
+              this.resources_.mutateElement(container, () => {});
```
Comment (cvializ) [inline extensions/amp-form/0.1/amp-form.js]: I think we should wrap the `appendChild` call in the `mutateElement` callback. It probably should have been that way to begin with.

Comment (alabiaga) [reply extensions/amp-form/0.1/amp-form.js]: done. thanks Carlos.

Comment (cvializ) [reply extensions/amp-form/0.1/amp-form.js]: @jridgewell to verify that that `appendChild` should be wrapped in `mutateElement`

File: extensions/amp-form/0.1/amp-form.js [hunk 2]
```diff
@@ -1045,21 +1045,21 @@ export class AmpForm {
...
               rendered.id = messageId;
               rendered.setAttribute('i-amphtml-rendered', '');
-              container.appendChild(rendered);
               const renderedEvent = createCustomEvent(
                   this.win_,
                   AmpEvents.DOM_UPDATE,
                   /* detail */ null,
                   {bubbles: true});
               container.dispatchEvent(renderedEvent);
+              this.resources_.mutateElement(container, () => {
```
Comment (choumx) [inline extensions/amp-form/0.1/amp-form.js]: Should we also wait to dispatch DOM_UPDATE event?

Comment (alabiaga) [reply extensions/amp-form/0.1/amp-form.js]: done. Strangely, the check type run fails when the dispatch event is inside the mutateElement call. 

Compiler issues for check-types.js:
extensions/amp-form/0.1/amp-form.js:1048: ERROR - actual parameter 1 of Resources$$module$src$service$resources_impl.prototype.mutateElement does not match formal parameter
found   : (Element|null)
required: Element
              this.resources_.mutateElement(container, () => {
                                            ^^^^^^^^^

I added a check type supress annotation to bypass it.

Comment (choumx) [reply extensions/amp-form/0.1/amp-form.js]: Use `devAssert(container)` instead.

Comment (alabiaga) [reply extensions/amp-form/0.1/amp-form.js]: done. thanks

**Comment by alabiaga** [comment]
thanks all!

**Pull request merged**